lib/enumerables.rb

```diff
@@ -0,0 +1,177 @@
+module Enumerable
+  # my_each
+  def my_each(&block)
+    arr = self
+    arr = arr.to_a if self.class == Range || arr.is_a?(Hash)
+    count = 0
+    if arr.is_a?(Hash)
+      while count < arr.length
+        yield arr[i][0], arr[i][1]
+        count += 1
+      end
+      self
+    elsif block
+      while count < arr.length
+        yield arr[count]
+        count += 1
+      end
+      self
+    else
+      "\#<Enumerator: #{self}:my_each>"
```

Comment:
This is not the correct way to return an enumerator. What you are doing here is actually returning a `string` object, not an `enumerator`.

To understand what I mean, try the following experiment, either by using the `irb` console or by writing an `.rb` script to run this:

```
[1, 2, 3].my_each.class
# This will output the class of the return value, which is currently String.
# It should output Enumerator.
```

In order to return a enumerator, you can replace this line with `arr.to_enum`.

After this, if you try the same experiment as before, you should now see `Enumerator` instead of `String`.